.github/scripts/license_years.py

```diff
@@ -0,0 +1,59 @@
+# SPDX-FileCopyrightText: 2024 Google LLC
+#
+# SPDX-License-Identifier: Apache-2.0
+import re
+from git import Repo
+
+# Initialize a git Repo object
+repo = Repo('.')
+git = repo.git
+
+# Get the last common ancestor of the current branch and the staging branch
+current_branch = repo.active_branch
+common_ancestor_hash = repo.git.merge_base('staging', current_branch.name)
+
+# Check if the current branch hash is the same as the staging hash
+if current_branch.commit.hexsha == common_ancestor_hash:
+  print("There are no commits between the current branch and the staging branch")
+  exit()
+
+# Get the commits that are in the current branch but not in main
+all_commits = repo.iter_commits()
+commits = []
+for commit in all_commits:
+  if commit.hexsha == common_ancestor_hash:
+    break
+  commits.append(commit.hexsha)
+
+# Check if the year in the license header is correct for each file in each commit
+all_valid = True
+for commit in commits:
+  obj = repo.commit(commit)
```

Comment:
So first you convert a commit object into a hexdigest, and now you're converting it to a commit object again? Just use commit objects throughout.